src/core/init/initialize_media_source.ts

```diff
@@ -147,185 +186,214 @@ export default function InitializeOnMediaSource(
+    new SegmentFetcherCreator<any>(transportPipelines, { lowLatencyMode,
+                                                         maxRetryOffline: offlineRetry,
+                                                         maxRetryRegular: segmentRetry });
 
-  // Create ABR Manager, which will choose the right "Representation" for a
-  // given "Adaptation".
+  /** Choose the right "Representation" for a given "Adaptation". */
   const abrManager = new ABRManager(adaptiveOptions);
 
-  // Create and open a new MediaSource object on the given media element.
-  // The MediaSource will be closed on unsubscription
-  const openMediaSource$ = openMediaSource(mediaElement).pipe(
-    subscribeOn(asapScheduler), // to launch subscriptions only when all
-    share());                   // Observables here are linked
+  /**
+   * Create and open a new MediaSource object on the given media element.
+   * The MediaSource will be closed on unsubscription.
+   */
+  const openMediaSource$ = openMediaSource(mediaElement)
+    .pipe(shareReplay()); // Cache value on success
 
-  // Send content protection data to the `EMEManager`
+  /** Send content protection data to the `EMEManager`. */
   const protectedSegments$ = new Subject<IContentProtection>();
 
-  // Create `EMEManager`, an observable which will handle content DRM
+  /** Create `EMEManager`, an observable which will handle content DRM. */
   const emeManager$ = openMediaSource$.pipe(
     mergeMap(() => createEMEManager(mediaElement, keySystems, protectedSegments$)),
     subscribeOn(asapScheduler), // to launch subscriptions only when all
     share());                   // Observables here are linked
 
-  // Translate errors coming from the media element into RxPlayer errors
-  // through a throwing Observable.
+  /**
+   * Translate errors coming from the media element into RxPlayer errors
+   * through a throwing Observable.
+   */
   const mediaError$ = throwOnMediaError(mediaElement);
 
-  const loadContent$ = observableCombineLatest([
-    openMediaSource$,
-    fetchManifest(url, undefined),
-    emeManager$.pipe(filter(isEMEReadyEvent), take(1)),
-  ]).pipe(mergeMap(([ initialMediaSource, initialManifestObj ]) => {
-    const { manifest } = initialManifestObj;
+  /**
+   * Emit when EME negociations that should happen before pushing any content
+   * have finished.
+   */
+  const waitForEMEReady$ = emeManager$.pipe(filter(isEMEReadyEvent),
+                                            take(1),
+                                            mapTo(true),
+                                            shareReplay()); // Cache value on success
+
+  /** Load and play the content asked. */
+  const loadContent$ = observableMerge(fetchManifest(url, undefined),
+                                       // Subscribe to both Observables without
+                                       // listening to prepare them
+                                       waitForEMEReady$.pipe(ignoreElements()),
```

Comment:
Do we really need to prepare them here ?

Reply:
If we do not:
  1. we will lose unnecessary time opening the MediaSource and waiting
  2. Most of all, we could miss the moment where the EME is ready and not start at all